### DISCUSSION TITLE
Uploading the releases to Github

### DISCUSSION THREAD

**Discussion by Geod24** [dlang/dmd#12150] [Ideas]
Hi everyone,
For a while now, I've been wanting to upload the releases to Github.

## Why

- Lower / remove the load on the s3 buckets we're using;
- There's built-in HTTPS;
- Likewise, Github is a trusted source for releases;
- We could further automate the release process;
- It would simplify https://github.com/dlang-community/setup-dlang and any Github-based workflow;

## What would change

Not much, it's just a matter of scripting the `wget` from `downloads.dlang.org` and pushing to the  releases.

## Where to ?

I was thinking of uploading the releases to DMD (this repository). Another option would be `installer` but DMD seems like the obvious choice, although the release also include tools, dub, etc...

Thoughts ?

**Comment by wilzbach** [comment]
I'm hugely in favor of this. We wanted to do this for a long time. The main reason this hasn't happened yet is because with the S3 the DLF has access to the traffic logs and statistics. However, these days Github has download statistics built-in, so I don't think it's a blocker.

And yep, definitely the dmd repository, because then we could even start setting up pipelines to automatically build the release (and resurrect the nightlies).

**Comment by MoonlightSentinel** [comment]
Possible implementation: #12491 ~~(currenlty WIP)~~
